README.md

```diff
@@ -8,77 +8,65 @@ npm install hypercore-protocol
+
 ## Usage
 
 ``` js
-var protocol = require('hypercore-protocol')
+const Protocol = require('hypercore-protocol')
 
 // create two streams with hypercore protocol
-var streamA = protocol({ id: 'a' })
-var streamB = protocol({ id: 'b' })
+const streamA = new Protocol(true)
+const streamB = new Protocol(false)
 
 // open two feeds specified by a 32 byte key
-var key = Buffer.from('deadbeefdeadbeefdeadbeefdeadbeef')
-var feed = streamA.feed(key)
-var remoteFeed = streamB.feed(key)
+const key = Buffer.from('deadbeefdeadbeefdeadbeefdeadbeef')
+const feed = streamA.open(key)
+const remoteFeed = streamB.open(key, {
+  // listen for data in remote feed
+  ondata (message) {
+    console.log(message.value.toString())
+  }
+})
 
 // add data to feed
 feed.data({ index: 1, value: '{ block: 42 }'})
 
-// listen data in remoteFeed
-remoteFeed.on('data', function (message) {
-  console.log(message.value.toString())
-})
-
 streamA.pipe(streamB).pipe(streamA)
 ```
 
 `output => { block: 42 }`
 
 ## API
 
-#### `var stream = protocol([options])`
+#### `const stream = new Protocol(initiator, [options])`
 
 Create a new protocol duplex stream.
 
 Options include:
 
 ``` js
 {
-  id: optionalPeerId, // you can use this to detect if you connect to yourself
-  live: keepStreamOpen, // signal to the other peer that you want to keep this stream open forever
-  ack: false, // Explicitly ask a peer to acknowledge each received block
-  userData: opaqueUserData // include user data that you can retrieve on handshake
   encrypt: true, // set to false to disable encryption if you are already piping through a encrypted stream
   timeout: 5000, // stream timeout. set to 0 or false to disable.
-  extensions: [], // names of extensions to use for replication. Must be sorted alphanumerically for handshaking to work
+  keyPair: { publicKey, secretKey }, // use this keypair for the stream authentication
+  onhandshake () { }, // function called when the stream handshake has finished
+  onopen (discoveryKey) { } // function called when the remote stream opens a feed you have not
 }
 ```
 
-If you don't specify a peer id a random 32 byte will be used.
-You can access the peer id using `p.id` and the remote peer id using `p.remoteId`.
-
-#### `var feed = stream.feed(key)`
+#### `stream.remotePublicKey`
 
-Signal the other end that you want to share a hypercore feed.
+The remotes public key.
 
-You can use the same stream to share more than one BUT the first feed shared
-should be the same one. The key of the first feed is also used to encrypt the stream using [libsodium](https://github.com/mafintosh/sodium-native#crypto_stream_xorcipher-message-nonce-key).
+#### `stream.publicKey`
 
-#### `var bool = stream.has(key)`
+Your public key.
 
-Returns true if the stream already has open a channel open for the given key and false if not.
+#### `const feed = stream.open(key, handlers)`
```

Comment:
What are the `handlers` here?

Reply:
Oh, are the various handlers documented below as `feed.handlers.*` ? If so, you might just mention that here.

Reply:
They are and I will